tt_eager/tt_dnn/op_library/topk/topk_op.cpp

```diff
@@ -8,8 +8,9 @@
 void TopK::validate(const std::vector<Tensor>& input_tensors) const {
     auto input_shape = input_tensors.at(0).get_legacy_shape();
     TT_FATAL(k == 32, fmt::format("K must be equal to 32, pad with -infinity if necessary"));
-    TT_FATAL(input_shape[-1] == 64, fmt::format("Input shape inner dim {} must be 64, pad with -infinity if necessary", input_shape[-1]));
-    TT_FATAL((input_shape[0] * input_shape[1] * input_shape[2]) == 32, "Input height must be 32");
+    TT_FATAL(input_shape[-1] >= 64, fmt::format("Input shape inner dim {} must be a multiple of 64, pad with -infinity if necessary", input_shape[-1]));
+    TT_FATAL((input_shape[-1] & (input_shape[-1] - 1)) == 0, fmt::format("Input shape inner dim {} must be a power of 2, pad with -infinity if necessary", input_shape[-1]));
+    TT_FATAL((input_shape[0] * input_shape[1] * input_shape[2]) % 32 == 0, fmt::format("Input height (combined input_shape[0-3]) {} must be a multiple of 32", input_shape[0] * input_shape[1] * input_shape[2]));
```

Comment:
does this TT_FATAL assume a tensor rank? if so can we add the tensor rank check to validate.

Reply:
In general, I think this works for ND:
- `-1` and `-2` for the last two dims
- multiply all dims up to last two dims for total batch
- `volume / -2` for the total "height" (this is what you want in line 13)

Reply:
Done